content/post/kubedb-v2024.02.14/index.md

```diff
@@ -0,0 +1,607 @@
+
+Apply the following YAML to deploy a ZooKeeper cluster with 3 replicas.
+
+```yaml
+apiVersion: kubedb.com/v1alpha2
+kind: ZooKeeper
+metadata:
+  name: zk-cluster
+  namespace: demo
+spec:
+  version: "3.9.1"
+  replicas: 3
+  storage:
+    resources:
+      requests:
+        storage: "1Gi"
+    storageClassName: "standard"
+    accessModes:
+    - ReadWriteOnce
+  terminationPolicy: "Halt"
+```
+
+#### supported versions
+`3.7.2`, `3.8.3` and `3.9.1`
+
+## Features and Improvements
+
+This release also contains some fixes, improvements and new features for existing database supports. Following are those in details.
+
+## Elasticsearch
+
+ElasticsearchDashboard, a custom resource used to provision Kibana for Elasticsearch cluster and Opensearch-Dashboards for Opensearch cluster has received a major change. Through this release we are bringing a breaking change by updating the `apiVersion` for this CR which has been updated to  `elasticsearch.kubedb.com/v1alpha1`.
+
+Here's a sample manifest to provision ElasticsearchDashboard. An Elasticsearch instance `es-cluster` is expected to be provisioned prior to deploying the following manifest.
+
+```yaml
+apiVersion: elasticsearch.kubedb.com/v1alpha1
+kind: ElasticsearchDashboard
+metadata:
+  name: es-cluster-dashboard
+  namespace: demo
+spec:
+  enableSSL: true
+  databaseRef:
+    name: es-cluster
+  terminationPolicy: WipeOut
+```
+
+## Kafka
+
+In this release, we have improved Kafka Health Checker. The health checker now ensures all kafka brokers are connected in a cluster, messages can be concurrently published and consumed via clients.
+Earlier, KubeDB managed Kafka had only two types of termination policy supports - `WipeOut` and `DoNotTerminate`. This release is bringing support for two more - `Halt` and `Delete`, providing the privilege of keeping PVCs and Secrets on cluster deletion.
+
+## Redis
+We have added support for initialization script in Redis. You can provide a `bash` or `lua` script through configmap and The script will be run at the start.
+
+Here is a sample configmap which has a `bash` script that creates two users in Redis nodes
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: redis-init-script
+  namespace: demo
+data:
+  script.sh: |
+
+    user1=$(cat auth/1/username)
+    pass1=$(cat auth/1/password)
+    user2=$(cat auth/2/username)
+    pass2=$(cat auth/2/password)
+
+
+    redis-cli ACL SETUSER "${user1}" on ">${pass1}" +@read ~* 
+    redis-cli ACL SETUSER "${user2}" on ">${pass2}"
+```
+Here's a sample Redis manifest that runs this script at the start.
+```yaml
+apiVersion: kubedb.com/v1alpha2
+kind: Redis
+metadata:
+  name: standalone-redis
+  namespace: demo
+spec:
+  version: 7.2.3
+  storage:
+    storageClassName: "standard"
+    accessModes:
+    - ReadWriteOnce
+    resources:
+      requests:
+        storage: 1Gi
+  init:
+    script:
+      projected:
+        sources:
+        - secret:
+            name: rd-auth1
+            items:
```

Comment:
Give the creation commands of these  secrets.

Reply:
@AbdullahAlShaad